src/esp/bindings/bindings.cpp

```diff
@@ -382,6 +348,73 @@ PYBIND11_MODULE(habitat_sim_bindings, m) {
   // ==== Observation ====
   py::class_<Observation, Observation::ptr>(m, "Observation");
 
+  m.attr("gpu_gpu_enabled") =
+#ifdef ESP_WITH_GPU_GPU
+      true;
+#else
+      false;
+#endif
+  py::class_<RenderingTarget, RenderingTarget::ptr>(m, "RenderingTarget")
+      .def("__enter__",
+           [](RenderingTarget::ptr self) {
+             self->renderEnter();
+             return self;
+           })
+      .def("__exit__", [](RenderingTarget::ptr self, py::object exc_type,
+                          py::object exc_value,
+                          py::object traceback) { self->renderExit(); })
+      .def(
+          "read_frame_rgba",
+          [](RenderingTarget& self,
+             Eigen::Ref<Eigen::Matrix<uint8_t, Eigen::Dynamic, Eigen::Dynamic,
+                                      Eigen::RowMajor>>& img) {
+            self.readFrameRgba(img.data());
+          },
+          py::arg("img").noconvert(),
+          R"(
+      Reads RGBA frame into passed img in uint8 byte format.
+
+      Parameters
+      ----------
+      img: numpy.ndarray[uint8[m, n], flags.writeable, flags.c_contiguous]
+           Numpy array array to populate with frame bytes.
+           Memory is NOT allocated to this array.
+           Assume that ``m = height`` and ``n = width * 4``.
+      )")
+      .def(
+          "read_frame_depth",
+          [](RenderingTarget& self,
+             Eigen::Ref<Eigen::Matrix<float, Eigen::Dynamic, Eigen::Dynamic,
+                                      Eigen::RowMajor>>& img) {
+            self.readFrameDepth(img.data());
+          },
+          py::arg("img").noconvert(), R"()")
+      .def(
+          "read_frame_object_id",
+          [](RenderingTarget& self,
+             Eigen::Ref<Eigen::Matrix<uint32_t, Eigen::Dynamic, Eigen::Dynamic,
+                                      Eigen::RowMajor>>& img) {
+            self.readFrameObjectId(img.data());
+          },
+          py::arg("img").noconvert(), R"()")
+#ifdef ESP_WITH_GPU_GPU
+      .def("read_frame_rgba_gpu",
+           [](RenderingTarget& self, size_t devPtr) {
+             self.readFrameRgbaGPU(reinterpret_cast<uint8_t*>(devPtr));
```

Comment:
Just wanted to draw attention to this wonkiness -- This is the pattern needed for passing pointers from python to c++.